libs/hdf-converters/src/netsparker-mapper.ts

```diff
@@ -139,87 +139,102 @@ function formatMessage(response: unknown): string {
   return text.join('\n');
 }
 export class NetsparkerMapper extends BaseConverter {
-  mappings: MappedTransform<ExecJSON.Execution, ILookupPath> = {
-    platform: {
-      name: 'Heimdall Tools',
-      release: HeimdallToolsVersion,
-      target_id: {path: 'netsparker-enterprise.target.url'}
-    },
-    version: HeimdallToolsVersion,
-    statistics: {
-      duration: null
-    },
-    profiles: [
-      {
-        name: 'Netsparker Enterprise Scan',
-        version: '',
-        title: {
-          path: 'netsparker-enterprise.target',
-          transformer: (input: unknown): string => {
-            return `Netsparker Enterprise Scan ID: ${_.get(
-              input,
-              'scan-id'
-            )} URL: ${_.get(input, 'url')}`;
-          }
-        },
-        maintainer: null,
-        summary: 'Netsparker Enterprise Scan',
-        license: null,
-        copyright: null,
-        copyright_email: null,
-        supports: [],
-        attributes: [],
-        depends: [],
-        groups: [],
-        status: 'loaded',
-        controls: [
-          {
-            path: 'netsparker-enterprise.vulnerabilities.vulnerability',
-            key: 'id',
-            id: {path: 'LookupId'},
-            title: {path: 'name'},
-            desc: {transformer: formatControlDesc},
-            impact: {
-              path: 'severity',
-              transformer: impactMapping(IMPACT_MAPPING)
-            },
-            tags: {
-              cci: {
-                path: 'classification',
-                transformer: (data: Record<string, unknown>) =>
-                  getCCIsForNISTTags(nistTag(data))
+  defineMappings(
+    toolname: string
+  ): MappedTransform<ExecJSON.Execution, ILookupPath> {
+    return {
+      platform: {
+        name: 'Heimdall Tools',
+        release: HeimdallToolsVersion,
+        target_id: {path: `${toolname}-enterprise.target.url`}
+      },
+      version: HeimdallToolsVersion,
+      statistics: {},
+      profiles: [
+        {
+          name: `${toolname.replace(/^./, (firstLetter) =>
```

Comment:
If we are controlling 'toolname' could we use toolname[0].toUpperCase() + toolname.substring(1). I don't think there is a performance issue, but regex to return the first character seems over-engineered. And would it be useful to save this as a const variable above the return since the upperCase is performed multiple times?

Reply:
I can put it into its own variable.  I find 'replace' as a function making more sense to me as to what's going on than arbitrary string concatenation.

Reply:
Approving with condition of change